includes/classes/ShowData.php

```diff
@@ -0,0 +1,55 @@
+<?php 
+class ShowData extends base
+{
+    protected $layout_data = [];
+    protected $configuration_data = [];
+    protected $prefix, $suffix,  $field_prefix, $field_suffix; 
+
+    public function __construct($lookup, $prefix = 'SHOW_', $suffix = '_INFO', $field_prefix = '_', $field_suffix = '') {
+      global $db; 
+
+      $this->prefix = $prefix; 
+      $this->suffix = $suffix; 
+      $this->field_prefix = $field_prefix; 
+      $this->field_suffix = $field_suffix; 
+
+      $sql = "SELECT products_type FROM " . TABLE_PRODUCTS . " WHERE products_id='" . $lookup . "'";
```

Comment:
Shouldn't the `$lookup` value be cast to an integer, quotes removed from the query (since the products_id is an integer) and a `LIMIT 1` be added?

Reply:
@lat9 in my testing on your comment, the script:
```
includes/init_includes/init_sanitize.php
```
is already sanitizing the `$_GET['products_id']` var, and redirecting you if the products id is not valid.  so i would say casting it to an integer is not necessary.

as far as the `LIMIT 1`, i would also say not a good idea, and specifically because `products_id` is the primary key.

Reply:
All good ideas.  Currently looking for directional feedback from @drbyte though.

Reply:
> 
> 
> @lat9 in my testing on your comment, the script:
> 
> ```
> includes/init_includes/init_sanitize.php
> ```
> 
> is already sanitizing the `$_GET['products_id']` var, and redirecting you if the products id is not valid. so i would say casting it to an integer is not necessary.

Yes, and that's one usage.  Since we're talking about a class-file, who knows what value could be specified by some other process.  Classes should be 'protectionist' about the inputs they're given.

> 
> as far as the `LIMIT 1`, i would also say not a good idea, and specifically because `products_id` is the primary key.

FWIW, I find the `LIMIT 1` to be descriptive in the reading of the code.